src/docfx/build/redirection/RedirectionMap.cs

```diff
@@ -111,6 +113,24 @@ void AddRedirections(Dictionary<string, SourceInfo<string>> items, bool redirect
                     }
                 }
             }
+
+            void CheckInvalidRedrectUrl()
+            {
+                var redirectUrls = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+
+                var publishUrls = buildFiles.Select(file => file.SiteUrl).Union(redirections.Select(redirection => redirection.SiteUrl)).ToHashSet();
```

Comment:
Union creates a hashset internally, there is already a ToHashSet at the end, so Concat is faster in this case
```suggestion
                var publishUrls = buildFiles.Select(file => file.SiteUrl).Concat(redirections.Select(redirection => redirection.SiteUrl)).ToHashSet();
```